packages/ember-glimmer/tests/integration/syntax/each-in-test.js

```diff
@@ -305,3 +305,29 @@ moduleFor('Syntax test: {{#each-in}} undefined path', class extends RenderingTes
     this.assertText('');
   }
 });
+
+moduleFor('Syntax test: {{#each-in}}  for array', class extends RenderingTest {
+  ['@test it iterate over array with `in` instead of walking over elements'](assert) {
+    let arr = [1, 2, 3];
+    arr.someKey = 'someKeyValue';
+
+    this.render(strip`
+      {{#each-in arr as |key value|}}
+        [{{key}}:{{value}}]
+      {{/each-in}}`, { arr: [] });
```

Comment:
Normally, you want to put the target test case first, so I would start with `{ arr }` instead.